warmrestart/warmRestartHelper.cpp

```diff
@@ -0,0 +1,511 @@
+         * NEW.
+         */
+        if (it->second != NEW)
+        {
+            it++;
+            continue;
+        }
+
+        for (auto const &fv : fvVector)
+        {
+            if (it->first == fv)
+            {
+                found = true;
+                break;
+            }
+        }
+
+        if (!found)
+        {
+            SWSS_LOG_INFO("Warm-Restart: Deleting transient best-path selection "
+                          "for entry %s\n", key.c_str());
+            it = fvMap.erase(it);
+        }
+        else
+        {
+            it++;
+        }
+    }
+}
+
+
+/*
+ * Reconciliation process takes place here.
+ *
+ * In a nutshell, the process relies on the following basic guidelines:
+ *
+ * - An element in the restorationMap with all its entries in the fvResMap showing
+ *   as STALE, will be eliminated from AppDB.
+ *
+ * - An element in the restorationMap with all its entries in the fvResMap showing
+ *   as CLEAN, will have a NO-OP associated with it -- no changes in AppDB.
+ *
+ * - An element in the restorationMap with all its entries in the fvResMap showing
+ *   as NEW, will correspond to a brand-new state, and as such, will be pushed to
+ *   AppDB.
+ *
+ * - An element in the restorationMap with some of its entries in the fvResMap
+ *   showing as CLEAN, will have these CLEAN entries, along with any NEW one,
+ *   being pushed down to AppDB.
+ *
+ * - An element in the restorationMap with some of its entries in the fvResMap
+ *   showing as NEW, will have these new entries, along with any CLEAN one,
+ *   being pushed to AppDB.
+ *
+ * - An element in the restorationMap with some/all of its entries in the
+ *   fvResMap showing as DELETE, will have these entries being eliminated
+ *   from AppDB.
+ *
+ */
+void WarmStartHelper::reconcile(void)
+{
+    SWSS_LOG_NOTICE("Initiating AppDB reconciliation process...");
+
+    assert(getState() == WarmStart::RESTORED);
+
+    /*
+     * Iterate through all the entries in the restorationMap and take note of the
+     * attributes associated to each.
+     */
+    auto it = m_restorationMap.begin();
+    while (it != m_restorationMap.end())
+    {
+        std::string key = it->first;
+        fvRestorationMap fvMap = it->second;
+
+        int totalRecElems, staleRecElems, cleanRecElems, newRecElems, deleteRecElems;
+        totalRecElems = staleRecElems = cleanRecElems = newRecElems = deleteRecElems = 0;
+
+        std::vector<FieldValueTuple> fvVector;
+
+        for (auto itMap = fvMap.begin(); itMap != fvMap.end(); )
+        {
+            totalRecElems++;
+
+            auto recElem = itMap->first;
+            auto recElemState = itMap->second;
+
+            if (recElemState == STALE)
+            {
+                itMap = fvMap.erase(itMap);
+                staleRecElems++;
+            }
+            else if (recElemState == CLEAN)
+            {
+                cleanRecElems++;
+                transformKFV(recElem, fvVector);
```

Comment:
Is the order of fields in fvRestorationMap guaranteed? 
I saw you put comment in fvComparator
```
/*
             * Notice that we are forced to iterate through all the tuples in the
             * right-vector given that the order of fields within a tuple is not
             * fully deterministic (i.e. 'left' could hold 'nh: 1.1.1.1 / if: eth0'
             * and 'right' could be 'if: eth0, nh: 1.1.1.1').
             */
```